File: libc/io.c
#include "io.h"
#include "defines.h"

u8 inportb(u16 port) {
  u8 ret;
  asmv("inb %1, %0" : "=a"(ret) : "dN"(port));
  return ret;
}

void outportb(u16 port, u8 data) {
  asmv("outb %1, %0" ::"dN"(port), "a"(data));
}

File: libgo/io.go
package utils

import  "C"

func Inportb(port uint16) uint8 {
	return uint8(C.inportb(C.uint16_t(port)))
}

func Outportb(port uint16, data uint8) {
	C.outportb(C.uint16_t(port), C.uint8_t(data))
}